prov/shm/src/smr_rma.c

```diff
@@ -161,7 +161,7 @@ ssize_t smr_generic_rma(struct smr_ep *ep, const struct iovec *iov,
 	    !(op_flags & FI_DELIVERY_COMPLETE)) {
 		smr_format_inline(cmd, iov, iov_count);
 	} else if (total_len <= SMR_INJECT_SIZE &&
-		   (smr_env.disable_cma || op == ofi_op_write) &&
+		   (ep->region->cma_cap == SMR_CMA_CAP_OFF || op == ofi_op_write) &&
```

Comment:
The cap check shouldn't be needed here because CMA doesn't affect whether we take the inject path or not (should be removed in the previous commit)

Reply:
Updated.